app/src/server/api/routers/combat.ts

```diff
@@ -552,6 +543,7 @@ export const combatRouter = createTRPCRouter({
       }
       // Determine battle background
       if (selectedAI) {
+        const background = await determineArenaBackground(ctx.drizzle, "arena");
```

Comment:
I'm wondering if we can figure out a way in which the queries for fetching background & user information (within initiateBattle) can be run in parallel within Promise.all.. every time we introduce sequential queries, we increase the function call time significantly, and the intiate battle call is one which is quite frequent.